src/Compressors/Gaussian.jl

```diff
@@ -0,0 +1,159 @@
+"""
+    Gaussian <: Compressor
+An implementation of the Gaussian compression method. This method forms a Gaussian sketch matrix 
+    with number of rows or number of columns setting as the compression dimension. 
+    
+# Mathematical Description
+    
+Let ``A`` be an ``m \\times n`` matrix that we want to compress. 
+
+If we want to compress ``A`` from the left (i.e., we reduce the number of rows), then 
+    we create a Gaussian sketch matrix, ``S``, with dimension ``s \\times m`` where 
+    ``s`` is the compression dimension that is supplied by the user. 
+    Each entry of ``S`` is generated independently following N(0, 1/s), namely, a 
+    Gaussian distribution with mean being zero and variance being 1 divided by the
+    compression dimension.
+    
+If ``A`` is compressed from the right, then we create a Gaussian sketch matrix, ``S``,
+    with dimension ``n \\times s``, where ``s`` is the compression dimension that 
+    is supplied by the user. 
+    Each entry of ``S`` is generated independently following N(0, 1/s), namely, a 
+    Gaussian distribution with mean being zero and variance being 1 divided by the 
+    compression dimension.
+
+# Fields
+
+- `cardinality::Type{<:Cardinality}`, the direction the compression matrix is intended to be 
+    applied to a target matrix or operator. Values allowed are `Left` or `Right`.
+- `compression_dim::Int64`, the target compression dimension. Referred to as ``s`` in the
+    mathematical description. 
+
+# Constructor
+
+    Gaussian(;cardinality=Left, compression_dim=2)
+
+## Arguments 
+- `cardinality::Type{<:Cardinality}`, the direction the compression matrix is intended to be 
+    applied to a target matrix or operator. Values allowed are `Left` or `Right`. By default 
+    `Left` is chosen. 
+- `compression_dim`, the target compression dimension. Referred to as ``s`` in the
+    mathemtical description. By default this is set to 2.
+
+## Returns 
+- A `Gaussian` object. 
+"""
+struct Gaussian <: Compressor
+    cardinality::Type{<:Cardinality}
+    compression_dim::Int64
+    # Check on the compression dimension
+    function Gaussian(cardinality, compression_dim)
+        if compression_dim < 0
```

Comment:
This only checks if `compression_dim` is negative, not if it is positive. Should be
`compression_dim < 1`